relay-config/src/config.rs

```diff
@@ -2273,25 +2273,15 @@ impl Config {
     }
 
     /// Redis servers to connect to, for rate limiting.
-    pub fn redis(&self) -> Option<(&RedisConnection, RedisConfigOptions)> {
-        let cpu_concurrency = self.cpu_concurrency();
-
-        let redis = self.values.processing.redis.as_ref()?;
-
-        let options = RedisConfigOptions {
-            max_connections: redis
-                .options
-                .max_connections
-                .unwrap_or(cpu_concurrency as u32 * 2)
-                .min(crate::redis::DEFAULT_MIN_MAX_CONNECTIONS),
-            connection_timeout: redis.options.connection_timeout,
-            max_lifetime: redis.options.max_lifetime,
-            idle_timeout: redis.options.idle_timeout,
-            read_timeout: redis.options.read_timeout,
-            write_timeout: redis.options.write_timeout,
-        };
+    pub fn redis(&self) -> Result<RedisPools, RedisError> {
```

Comment:
Returning `RedisPools` here introduces some coupling between `relay_config` and `relay_redis`. I previously had an auxiliary type here that kept everything local, but I think it's not worth the additional indirection and ceremony.